import-scripts/annotation_suite/merge_mafs.py

```diff
@@ -0,0 +1,122 @@
+    """ Generates a merged MAF given a list of input MAFs. """
+    merged_header = merge_headers(input_mafs)
+    sequenced_samples = set()
+
+    rows_to_write = []
+    for input_maf in input_mafs:
+        print('Loading data from ' + input_maf)
+        with open(input_maf, "r") as maf:
+            header_processed = False
+            file_header = get_file_header(input_maf)
+            for line in maf:
+                # add comment processing later (sequenced samples header?)
+                if line.startswith("#"):
+                    continue
+                if not header_processed:
+                    header_processed = True
+                    continue
+                # map row values to current header columns
+                mapped_row = dict(zip(file_header, list(map(lambda x: process_datum(x), line.split("\t")))))
+                sample_id = mapped_row[TUMOR_SAMPLE_BARCODE_COLUMN]
+
+                if sample_id:
+                    sequenced_samples.add(sample_id)
+                # full record for the merged MAF (if mapped_row does not contain a column in merged_header, it is blank)
+                normalized_row = list(map(lambda x: mapped_row.get(x, ""), merged_header))
+                rows_to_write.append("\t".join(normalized_row))
+
+    with open(output_maf_filename, "w") as output_maf:
+        output_maf.write("#sequenced_samples: %s" % (" ".join(sequenced_samples)))
+        output_maf.write("\n")
+        output_maf.write("\t".join(merged_header))
+        output_maf.write("\n")
+        output_maf.write("\n".join(rows_to_write))
+        output_maf.write("\n")
+    print('Merged MAF written to: %s' % (output_maf_filename))
+
+def verify_input_mafs(input_mafs):
+    for maf in input_mafs:
+        if not os.path.isfile(maf):
+            print("Could not find MAF by filename: %s" % (maf))
+            exit(1)
+
+def usage(parser):
+    parser.print_usage()
+    sys.exit(2)
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("-i", "--input-mafs-list", action = "store", dest = "input_mafs_list", help = "comma-delimited list of MAFs to merge", required = False)
+    parser.add_argument("-d", "--input-mafs-directory", action = "store", dest = "input_mafs_directory", help = "directory containing all MAFs to merge", required = False)
+    parser.add_argument("-o", "--output-maf", action = "store", dest = "output_maf", help = "output filename for merged MAF", required = True)
+
+    args = parser.parse_args()
+    input_mafs_list = args.input_mafs_list
+    input_mafs_directory = args.input_mafs_directory
+    output_maf = args.output_maf
+
+    if (input_mafs_list and input_mafs_directory) or (not input_mafs_list and not input_mafs_directory):
+        print("Please choose only one of the following options when running script:  --input-mafs-list | --input-mafs-directory")
+        usage(parser)
+
+    if input_mafs_list:
+        input_mafs = input_mafs_list.split(",")
```

Comment:
maybe map a "trim()" call to each member of this list (in case a comma separated list contains spaces and is in quotes)